### ISSUE TITLE
Spyder's `savefig` differs from plain terminal's

### ISSUE THREAD

**Issue by OverLordGoldDragon** [spyder-ide/spyder#19672]
How does Spyder change `plt.savefig()` behavior? Comparison below. This creates major discrepancy between what Spyder generates and other applications (e.g. Jupyer, `sphinx-gallery`), while I need them to be same. Can Spyder be made to match terminal `python example.py`?

<hr>

```python
import matplotlib.pyplot as plt

plt.plot([1, 2])
plt.gcf().set_size_inches(6, 4)
plt.gcf().set_dpi(72)
plt.gcf().savefig("img.png")
```

**Command prompt**:

![image](https://user-images.githubusercontent.com/16495490/193394659-ddf32ad2-daa5-45b4-8993-de891db7efb7.png)

**Spyder**

![image](https://user-images.githubusercontent.com/16495490/193394691-d58b0b2b-6062-42bb-8e98-ad98d7a3741b.png)

with

![image](https://user-images.githubusercontent.com/16495490/193394687-9e491dbb-dc03-4905-81c7-24c228510445.png)

**Comment by dalthviz** [comment]
Hi @OverLordGoldDragon thanks for the feedback! I was able to reproduce this. Checking a little bit I think this could be related with other values like font size that get setup on the kernel as default values:

https://github.com/spyder-ide/spyder-kernels/blob/9697d753267e9a27998ba72a82046413181be9e9/spyder_kernels/console/start.py#L145-L160

Maybe those values should be also exposed under the preferences page @ccordoba12 ?

**Comment by ccordoba12** [comment]
I think the value that we should expose to make both plots match is `font.size`. @dalthviz. could you check if increasing it to 11 or 12 makes things work ok?

**Comment by dalthviz** [comment]
After checking, seems like the font size has some effect on the file size but not in its dimentions. However, if I change the backend to `Qt5` I can get the same .png from Spyder and from an Anaconda prompt:

Running the script selecting the `Qt5` backend from Spyder:

![img_spyder](https://user-images.githubusercontent.com/16781833/195650370-261037eb-5f8b-46f8-8f5f-e38a34e5aa52.png)

Running the script from an Anaconda prompt:

![img_interpreter](https://user-images.githubusercontent.com/16781833/195650427-d5fc854a-7b8d-47f8-84df-c49e3e9ce55d.png)

I guess the default backend when running from a terminal is not `inline` but `Qt5`?

To use the inline backend in both Spyder and the Anaconda prompt/terminal I had to change the script to ensure what backend was being used but also set the config values through `rcParams` to match the ones set on `spyder-kernels`:

```python
import matplotlib
import matplotlib.pyplot as plt

# Set matplotlib backend
matplotlib.use('module://matplotlib_inline.backend_inline')
# Check matplotlib backend
print(matplotlib.get_backend())

# Set config params
matplotlib.rcParams['figure.figsize'] = (6, 4)
matplotlib.rcParams['figure.dpi'] = 72
matplotlib.rcParams['font.size'] = 10
matplotlib.rcParams['figure.subplot.bottom'] = 0.125

plt.plot([1, 2])
plt.gcf().savefig("img.png")
```

Inline from Spyder:

![img_spyder_inline](https://user-images.githubusercontent.com/16781833/195663847-0e0a4a7d-b673-40d2-9c9c-2f81ee17af24.png)

Running the changed script from an Anaconda prompt:

![img_interpreter_inline](https://user-images.githubusercontent.com/16781833/195663843-37faad54-19ee-46aa-94cc-5d358c2f93f4.png)

If I don't change the font size and subplots config values in the script, I get an image with the plot with the same dimentions using Spyder, but the file size differs and the plot is a little bit down if running from an Anaconda prompt/terminal:

```python
import matplotlib
import matplotlib.pyplot as plt

matplotlib.use('module://matplotlib_inline.backend_inline')
print(matplotlib.get_backend())
matplotlib.rcParams['figure.figsize'] = (6, 4)
matplotlib.rcParams['figure.dpi'] = 72
# matplotlib.rcParams['font.size'] = 10
# matplotlib.rcParams['figure.subplot.bottom'] = 0.125
plt.plot([1, 2])
plt.gcf().savefig("img_spyder_inline_def_font.png")
``` 

Running the changed script from Spyder:

![img_spyder_inline_def_font](https://user-images.githubusercontent.com/16781833/195665276-074ebf11-8dff-4664-b008-78b2451f4726.png)


Running the changed script from an Anaconda prompt without font size and subplots configs (I thing the default font size used here is 12 and the subplots bottom is 0.11):

![img_interpreter_inline_def_font](https://user-images.githubusercontent.com/16781833/195664516-663e845d-fb80-421f-b8c1-724d142f37d4.png)

**Comment by OverLordGoldDragon** [comment]
@dalthviz Thanks for this, all confirmed plus, below seems to reproduce non-Spyder in Spyder:

```python
# taken from Anaconda prompt's values
matplotlib.rcParams['figure.figsize'] = (6.4, 4.8)
matplotlib.rcParams['figure.dpi'] = 100
matplotlib.rcParams['font.size'] = 10
matplotlib.rcParams['figure.subplot.bottom'] = 0.11
```
Though one thing's that's not reproduced is that Spyder somehow always shows the plots as pretty (by resizing them appropriately), while the browser for some reason only does it if set to 80% zoom (Firefox, Edge). But that's likely a non-Spyder issue. (as an aside, if anyone has a reference on the 80% thing, it's appreciated)

Feel free to close.

**Comment by ccordoba12** [comment]
Yeah, let's keep it and take the opportunity to add those two config values to our Preferences. However, we'd need to do it in a minor release (i.e. 5.4, 5.5, or even 6.0) given that it requires changes in both Spyder and Spyder-kernels.

Do you have time to work on it for this release?

**Comment by dalthviz** [comment]
I don't think so. I guess this then should go to the v5.x millestone or to the v6.0alpha1 one?

**Comment by ccordoba12** [comment]
Ok, no problem. I moved it to Spyder 6 because is not critical.

**Comment by jitseniesen** [comment]
@dalthviz Can I take over this issue?

**Comment by dalthviz** [comment]
Sure @jitseniesen thanks!
